surface/include/pcl/surface/advancing_front_utils.h

```diff
@@ -0,0 +1,310 @@
+      double d4343 = (p4 - p3).dot ((p4 - p3));
+
+      double denom = d2121 * d4343 - d4321 * d4321;
+      if (denom < 1.0e-8) // The lines are parallel
+      {
+        DistPoint2LineResults temp, dist;
+        result.parallel = true;
+
+        dist = distPoint2Line (p1, p2, p3);
+        result.mu[0] = dist.mu;
+
+        temp = distPoint2Line (p1, p2, p4);
+        if (temp.mu < dist.mu)
+          result.mu[0] = temp.mu;
+
+        result.p[0] = p1 + result.mu[0] * (p2 - p1);
+        temp = distPoint2Line (p3, p4, result.p[0]);
+
+        result.mu[1] = temp.mu;
+        result.p[1] = p3 + result.mu[1] * (p4 - p3);
+      }
+      else
+      {
+        result.mu[0] = (d1343 * d4321 - d1321 * d4343) / denom;
+        result.mu[1] = (d1343 + result.mu[0] * d4321) / d4343;
+
+        if (result.mu[0] > 1.0)
+          result.mu[0] = 1.0;
+        else if (result.mu[0] < 0.0)
+          result.mu[0] = 0.0;
+
+        if (result.mu[1] > 1.0)
+          result.mu[1] = 1.0;
+        else if (result.mu[1] < 0.0)
+          result.mu[1] = 0.0;
+
+        result.p[0] = p1 + result.mu[0] * (p2 - p1);
+        result.p[1] = p3 + result.mu[1] * (p4 - p3);
+      }
+
+      result.d = (result.p[1] - result.p[0]).norm ();
+      return result;
+    }
+
+    struct IntersectionLine2PlaneResults
+    {
+      Eigen::Vector3f points[2]; /**< \brief The points defining the line. */
+      Eigen::Vector3f w;         /**< \brief The vector defining the line direction. */
+      Eigen::Vector3f origin;    /**< \brief The origin of the plane. */
+      Eigen::Vector3f u;         /**< \brief The vector defining the planes relative u directions. */
+      Eigen::Vector3f v;         /**< \brief The vector defining the planes relative v directions. */
+      Eigen::Vector3f p;         /**< \brief The point of intersection p = origin + x * u + y * v  || p = points[0] + w * lw */
+      float mw;                  /**< \brief The parametric coeff defining the intersection location on the line. */
+      float mu;                  /**< \brief The parametric plane u coeff of intersetion. */
+      float mv;                  /**< \brief The parametric plane v coeff of intersetion. */
+      bool parallel;             /**< \brief Indicate whether the line is parallel to the plane. */
+    };
+
+    /**
+      * \brief Find the intersection between a plane and line
+      * \param[in] p1     The origin point of the line
+      * \param[in] p2     The terminating point of the line
+      * \param[in] origin The origin of the plane
+      * \param[in] u      The vector defining the u direction for the plane
+      * \param[in] v      The vector defining the v direction for the plane
+      * \return IntersectionLine2PlaneResults
+      */
+    IntersectionLine2PlaneResults
+    intersectionLine2Plane (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2, const Eigen::Vector3f &origin, const Eigen::Vector3f &u, const Eigen::Vector3f &v)
```

Comment:
There is a function to compute line with plane intersection in `common` module "intersections.h" file. Can we reuse it?

Reply:
I only see lineWithLineIntersection and planeWithPlaneIntersection but no lineWithPlaneIntersection. Is it located somewhere else?

Reply:
Sorry, my bad. Can we add your function there, then?

Reply:
OK good. I will move the function to the intersection.h file.